deepview_profile/analysis/runner.py

```diff
@@ -4,32 +4,41 @@
 from deepview_profile.analysis.session import AnalysisSession
 from deepview_profile.nvml import NVML
 from deepview_profile.utils import release_memory
+import weakref
 
-
-def analyze_project(project_root, entry_point, nvml):
+def analyze_project(project_root, entry_point, nvml, ddp_request):
     session = AnalysisSession.new_from(project_root, entry_point)
+    
     release_memory()
-
     print("analyze_project: running measure_breakdown()")
     yield session.measure_breakdown(nvml)
+    
     release_memory()
-
     print("analyze_project: running measure_throughput()")
     yield session.measure_throughput()
+    
     release_memory()
-
     print("analyze_project: running deepview_predict()")
     yield session.habitat_predict()
+    
     release_memory()
-
     print("analyze_project: running measure_utilization()")
     yield session.measure_utilization()
-    release_memory()
 
+    release_memory()
     print("analyze_project: running energy_compute()")
     yield session.energy_compute()
-    release_memory()
 
+    if ddp_request:
+        release_memory()
+        print("analyze_project: running ddp_computation()")
+        yield session.ddp_computation()
+    
+    # release object session (less gpu memory consumption)
+    release_memory()
+    weakref.finalize(session,print,"session object destroyed")
```

Comment:
Please format this line with a code formatter.

Reply:
Done